channeld/channeld.c

```diff
@@ -2915,6 +2915,43 @@ static size_t calc_weight(enum tx_role role, const struct wally_psbt *psbt)
 	return weight;
 }
 
+/* Get the remote node amount after the splice */
+static struct amount_msat remote_balance(struct peer *peer,
+					    enum tx_role our_role,
+					    const struct wally_psbt *psbt,
+					    int chan_output_index,
+					    int chan_input_index)
+{
+	s64 relative_remote_val;
+	/* Current value of the remote channel balance */
+	struct amount_msat push_value;
+
+	/* We calculcate the `push_value` to send to the
+	 * hsmd, that is the remote amount in the channel
+	 * after the splice. */
+	switch (our_role) {
+	case TX_INITIATOR:
+		push_value = peer->channel->view[REMOTE].owed[REMOTE];
+		relative_remote_val = peer->splicing->accepter_relative;
+		break;
+	case TX_ACCEPTER:
+		push_value = peer->channel->view[REMOTE].owed[REMOTE];
```

Comment:
this is the same as line 2934

Reply:
ah yes I can move this out of the switch, but `peer->channel->view[REMOTE].owed[REMOTE]` should be always the remote balance of the channel, as I learned from this comment https://github.com/vincenzopalazzo/lightning/blob/master/common/initial_channel.h#L25

So in this case we need to know the remote balance, so only the splice information are a in different places.

I am missing somethings?

Reply:
I would still move this line out of the switch - I don't see that the duplication adds any readability, maybe even reduces it.

Reply:
I agree 💯